pkg/mkverification-raw-efi/vmcap.go

```diff
@@ -0,0 +1,387 @@
+// Copyright (c) 2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"encoding/binary"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// For more information chekout out https://man7.org/linux/man-pages/man4/msr.4.html
+// and https://cdrdv2.intel.com/v1/dl/getContent/671200 (Appending A -VMX capability
+// reporting facility, on page 4483)
+
+const (
+	basic             = 0x480
+	pinBasedCtls      = 0x481
+	procBasedCtls     = 0x482
+	exitCtls          = 0x483
+	entryCtls         = 0x484
+	miscCtls          = 0x485
+	procBasedCtls2    = 0x48B
+	eptVPIDCap        = 0x48C
+	truePinBasedCtls  = 0x48D
+	trueProcBasedCtls = 0x48E
+	trueExitCtls      = 0x48F
+	trueEntryCtls     = 0x490
+	vmFunc            = 0x491
+	procBasedCtls3    = 0x492
+)
+
+type msr struct {
+	f *os.File
+}
+
+func newMsr() (*msr, error) {
+	f, err := os.OpenFile("/dev/cpu/0/msr", os.O_RDONLY, 0)
```

Comment:
I do not see the file is ever closed. I understand, this is a small one-run tool, which does its job and exits, so fd leakage will be never observed, but code usually tends to spread by copy-pasting, and I will not be surprised that it will appear in some long-running daemon someday :)